jobs/validate/calico-spec

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 # shellcheck disable=SC2034,SC1090
 
-set -ex
+set -x
```

Comment:
This change seems kinda scary. It means that any of these juju or tigera_aws.py commands could fail and we would just... ignore the error and move on, right?

Reply:
hmm... well, yes that's true.  But I imagine it would result in a later test failure.  The current situation is that if any of those commands fail, the only commands that execute post failure are the `ci::cleanup` trap in ci.bash. 

This results in a cleaned up model but no test reporting on jenkaas.

Reply:
maybe there's a good way to "wrap and test" all the potential failures better than `set -e` in the whole file

Reply:
Yeah, I guess this is better than how it was. Better to have a slow test run and a confusing failure report than to have no report at all.

I do think it would be better if we could find a way to fix this while preserving the fail-fast nature of the script, but it seems like these jobs and the report handling aren't really set up to do it well. I think we would either have to keep `set -e` and report run failures in a trap, or remove `set -e` and add a lot of return code checks to the script. Or rewrite the whole thing in Python.

Reply:
i went with the approach of wrapping the "must work" methods with `( ... )` and then checking the whole exit code of that bash block.  If there is an error, we'll call that a "failed deployment"

Reply:
That sounds good to me. Thanks.